tests/examples/source/devfiles/springboot/devfile-with-missing-projectType-and-language-metadata.yaml

```diff
@@ -20,7 +20,7 @@ components:
       mountSources: true
   - name: runtime
     container:
-      image: quay.io/eclipse/che-java11-maven:next
+      image: registry.access.redhat.com/ubi8/openjdk-11:latest
```

Comment:
/hold cancel

Unholding as it is still fine, but we can consider updating `mountSources` in a separate PR.

Reply:
The error I have seen (and which has been fixed by changing the image) was occuring during the build command.

Reply:
Yes, for the build, I think it was a DNS resolution error (from what I can see in the [job logs](https://storage.googleapis.com/origin-ci-test/logs/periodic-ci-redhat-developer-odo-main-v4.13-integration-e2e-periodic/1648839060115951616/build-log.txt)):
```
 [odo]  ✗  Building your application in container (command: defaultbuild) [2s]
  [odo] [INFO] Scanning for projects...
  [odo] Downloading from central: https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-starter-parent/2.3.5.RELEASE/spring-boot-starter-parent-2.3.5.RELEASE.pom
  [odo] [ERROR] [ERROR] Some problems were encountered while processing the POMs:
  [odo] [FATAL] Non-resolvable parent POM for com.example:demo:0.0.1-SNAPSHOT: Could not transfer artifact org.springframework.boot:spring-boot-starter-parent:pom:2.3.5.RELEASE from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-starter-parent/2.3.5.RELEASE/spring-boot-starter-parent-2.3.5.RELEASE.pom and 'parent.relativePath' points at no local POM @ line 5, column 10
  [odo]  @ 
  [odo] [ERROR] The build could not read 1 project -> [Help 1]
  [odo] [ERROR]   
  [odo] [ERROR]   The project com.example:demo:0.0.1-SNAPSHOT (/projects/pom.xml) has 1 error
  [odo] [ERROR]     Non-resolvable parent POM for com.example:demo:0.0.1-SNAPSHOT: Could not transfer artifact org.springframework.boot:spring-boot-starter-parent:pom:2.3.5.RELEASE from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-starter-parent/2.3.5.RELEASE/spring-boot-starter-parent-2.3.5.RELEASE.pom and 'parent.relativePath' points at no local POM @ line 5, column 10: Unknown host repo.maven.apache.org: Name or service not known -> [Help 2]
```

But once that passes, the `run` command (which runs the `spring-boot:run` goal) cannot run if the sources are not mounted (or if the Maven settings in the container image do not indicate how to resolve the `spring-boot` plugin).
I'll create a separate PR to mount the sources so that this can run properly..